gems/aws-sdk-core/lib/aws-sdk-core.rb

```diff
@@ -1,13 +1,56 @@
 # frozen_string_literal: true
 
+require 'hearth'
 require 'aws-sigv4'
 
 # Namespace for AWS Ruby products
 module AWS
   # Namespace for AWS SDK gems
   module SDK
     # Namespace for AWS::SDK Core components
-    module Core; end
+    module Core
```

Comment:
We already open up AWS::SDK::Core below. We define this here first so that requiring doesn't have issues. Why do this twice?

Reply:
Because any code in the requires below that needs to switch on features of crt/sso/sts needs these defined BEFORE they are required (eg, the checksums code must be required AFTER these are defined)

Reply:
Can we just move that require statement to the end of the first definition then?